<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageTaskScriptingWidget</class>
 <widget class="QWidget" name="ImageTaskScriptingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>300</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <widget class="QPushButton" name="editShapeScriptsButton">
     <property name="text">
      <string>Edit Shape Creation/Mutation Scripts</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="customScriptsGroupBox">
     <property name="title">
      <string>Add And Edit Custom Scripts</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="addBeforeStepEvent">
            <property name="text">
             <string>Before Step</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addBeforeStepPresetScriptsComboBox">
            <property name="currentText">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="addAfterStepEvent">
            <property name="text">
             <string>After Step</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addAfterStepPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="addStopConditionButton">
            <property name="text">
             <string>Stop Condition</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addStopConditionPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <widget class="QPushButton" name="onStopConditionMetEvent">
            <property name="text">
             <string>On Stop Condition Met</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onStopConditionMetEventComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_18">
          <item>
           <widget class="QPushButton" name="addShapePreconditionButton">
            <property name="text">
             <string>Add Shape Precondition</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addShapePreconditionPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="addBeforeAddShapeEvent">
            <property name="text">
             <string>Before Add Shape</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addBeforeAddShapePresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="addAfterAddShapeEvent">
            <property name="text">
             <string>After Add Shape</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="addAfterAddShapePresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QPushButton" name="onScenePenInputEvent">
            <property name="text">
             <string>On Scene Pen Input</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onScenePenInputEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QPushButton" name="onPenProximityEnterEvent">
            <property name="text">
             <string>On Pen Proximity Enter</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onPenProximityEnterEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QPushButton" name="onPenProximityExitEvent">
            <property name="text">
             <string>On Pen Proximity Exit</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onPenProximityExitEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <widget class="QPushButton" name="onKeyDownEvent">
            <property name="text">
             <string>On Key Down</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onKeyDownEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QPushButton" name="onKeyUpEvent">
            <property name="text">
             <string>On Key Up</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onKeyUpEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_13">
          <item>
           <widget class="QPushButton" name="onMouseDownEvent">
            <property name="text">
             <string>On Mouse Down</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onMouseDownEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_15">
          <item>
           <widget class="QPushButton" name="onMouseUpEvent">
            <property name="text">
             <string>On Mouse Up</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onMouseUpEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QPushButton" name="onMouseMoveEvent">
            <property name="text">
             <string>On Mouse Move</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onMouseMoveEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_16">
          <item>
           <widget class="QPushButton" name="onMouseWheelEvent">
            <property name="text">
             <string>On Mouse Wheel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onMouseWheelEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QPushButton" name="onUpdateEvent">
            <property name="text">
             <string>On Update</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="onUpdateEventPresetScriptsComboBox"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="customScriptsEditorLayout"/>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rescanScriptsButton">
        <property name="text">
         <string>Rescan Scripts</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearScriptsButton">
        <property name="text">
         <string>Clear Scripts</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
